packages/eslint-plugin/src/rules/space-before-function-paren.ts

```diff
@@ -75,16 +75,18 @@ export default util.createRule<Options, MessageIds>({
       node:
         | TSESTree.ArrowFunctionExpression
         | TSESTree.FunctionDeclaration
-        | TSESTree.FunctionExpression,
+        | TSESTree.FunctionExpression
+        | TSESTree.TSAbstractMethodDefinition,
     ): boolean {
-      if (node.id) {
+      if (Reflect.get(node, 'id')) {
```

Comment:
why are you using `Reflect` here and other places?

Reply:
because id not exists in the TSESTree.TSAbstractMethodDefinition type

Reply:
instead of `Reflect`, please use an explicit check like `'id' in node && node.id != null`.
There's no need to use

Reply:
ok you who's boss